### ISSUE TITLE
how to evaluate?

### ISSUE THREAD

**Issue by zhaobaozi** [wvangansbeke/Sparse-Depth-Completion#15]
Hi!
I  am confuse how to evaluate the dataset?I see some code can do that in main.py,but in "elif args.evaluate......validate(valid_selection_loader,model,criterion_global,criterion_local) " ,i meet a error,that is the "criterion_global" is not defined,how can i do?
thanks! 
@wvangansbeke

**Comment by wvangansbeke** [comment]
Hey @zhaobaozi,

Yes you're right. I changed some names afterwards and forgot to change it there. However it should be fairly easy to correct, because it needs to be consistent with this line [here](https://github.com/wvangansbeke/Sparse-Depth-Completion/blob/d430121719c5bf0d00a8c00feecd29e232d3c4dd/main.py#L336). I will correct it later, but this will not affect anything else.

Thanks for letting me know.
Best,
Wouter

**Comment by zhaobaozi** [comment]
@wvangansbeke 
Thanks for your reply,i have finish this problem.
But when i want to train the model (use pretrin model),I meet another problem.
`        else:
            check = torch.load('external_mod.pth.tar')
            lowest_loss_load = check['loss']
            target_state = model.state_dict()
            for name, val in check['state_dict'].items():
                if name not in target_state:
                    continue
                try:
                    target_state[name].copy_(val)
                except RuntimeError:
                    continue
            print("=> loaded EXTERNAL checkpoint with best rmse {}"
                    .format(lowest_loss_load))`

I can't see  the 'target_state' use in later training,just assignment.
thanks!

**Comment by yan2019626** [comment]
Hello! Thanks for sharing this great work. I get an error 'cannot identify files../Rgb/train.../.txt  or .bin' when training. I do not know why it happens. When I delete all .bin and .txt files, the training can run.

**Comment by yan2019626** [comment]
> Hi!
> I am confuse how to evaluate the dataset?I see some code can do that in main.py,but in "elif args.evaluate......validate(valid_selection_loader,model,criterion_global,criterion_local) " ,i meet a error,that is the "criterion_global" is not defined,how can i do?
> thanks!
> @wvangansbeke

Can you show me your specific dataset structure? It gets abnormal RMSE when training. I think my training dataset is not delivered to the network correctly.
 @wvangansbeke

**Comment by wvangansbeke** [comment]
Hi @yan2019626,

You can find the network structure in the README. Feel free to adapt `Datasets/Kitti_loader.py`. if it's different from your dataset structure. It should be straightforward. 

Best,
Wouter
